### PR TITLE
CAMEL-15769: disable split default delimiter

### PR DESCRIPTION
**Pull request by srcarrier**
- [x] Make sure there is a [JIRA issue](https://issues.apache.org/jira/browse/CAMEL) filed for the change (usually before you start working on it).  Trivial changes like typos do not require a JIRA issue.  Your pull request should address just this issue, without pulling in other changes.
- [x] Each commit in the pull request should have a meaningful subject line and body.
- [x] If you're unsure, you can format the pull request title like `[CAMEL-XXX] Fixes bug in camel-file component`, where you replace `CAMEL-XXX` with the appropriate JIRA issue.
- [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [x] Run `mvn clean install -Psourcecheck` in your module with source check enabled to make sure basic checks pass and there are no checkstyle violations. A more thorough check will be performed on your pull request automatically.
Below are the contribution guidelines:
https://github.com/apache/camel/blob/master/CONTRIBUTING.md

This PR addresses CAMEL-15769

I followed an implementation similar to org.apache.camel.processor.RecipientList#IGNORE_DELIMITER_MARKER, as cited by Claus in the referenced Jira issue.

This is my first camel contribution, so I appreciate any guidance on the fix or any contribution steps I might have missed.
Thanks in advance

### CODE REVIEW

File: core/camel-core-model/src/main/java/org/apache/camel/model/SplitDefinition.java [hunk 1]
```diff
@@ -517,6 +520,14 @@ public void setExpression(ExpressionDefinition expression) {
         super.setExpression(expression);
     }
 
+    public String getDelimiter() {
+        return delimiter;
+    }
+
+    public void setDelimiter(String delimiter) {
```
Comment (davsclaus) [inline core/camel-core-model/src/main/java/org/apache/camel/model/SplitDefinition.java:527]: We need documentation for this new option. See elsewhere in this class for those fluent methods that have javadoc. Then add a fluent method for delimiter with javadoc

Comment (srcarrier) [reply core/camel-core-model/src/main/java/org/apache/camel/model/SplitDefinition.java:540]: Thanks Claus - see latest commit 50d57a986c6fe748313a8d195060e74e4cb3fc51
I wasn't sure how to address the build failure reported with the previous commit _(sorry, I didn't grab the error msg prior to pushing the latest commit)_. A new build is in progress now. If it fails, I may need some pointers on how to resolve.
Thanks in advance

**Review by davsclaus — APPROVED** [review]

**Comment by davsclaus** [comment]
Ah those json files just need to be regenerated. Let me try that

**Comment by srcarrier** [comment]
oh, sorry Claus - I managed to generate the json files locally and committed them.
Hopefully I didn't step on your toes there.

**Pull request merged**